sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/orc/OrcFileFormat.scala

```diff
@@ -188,6 +188,9 @@ class OrcFileFormat
         if (enableVectorizedReader) {
           val batchReader = new OrcColumnarBatchReader(
             enableOffHeapColumnVector && taskContext.isDefined, copyToSpark, capacity)
+          val iter = new RecordReaderIterator(batchReader)
+          Option(TaskContext.get()).foreach(_.addTaskCompletionListener(_ => iter.close()))
+
           batchReader.initialize(fileSplit, taskAttemptContext)
```

Comment:
According to the reported case, the ORC file is opened here. 
But, it seems that the task is killed, `TaskKilled (Stage cancelled)`, before registering its listener during `initBatch`.

Reply:
@cloud-fan and @gatorsmile . Could you take a look this?
For ORC library, it looks okay when we call `close` correctly.

Reply:
@dongjoon-hyun Thanks for this fix! My question is how do we know if `close` is not called and is called now? Have you verified it?

Reply:
Because I tried to verify it manually in local, seems `close` is called before this change. Maybe I miss something or this is environment depending.